screens/CreateEventScreen.jsx

```diff
@@ -0,0 +1,414 @@
+import { editProfile, uploadPicture } from '../api/firebaseMethods';
+import { follow, unfollow } from '../api/firebaseMethods'
+import { useUser, useUserUpdate } from '../hooks/UserContext';
+
+export default function CreateEventScreen({ navigation }) {
+  const [wordCount, setWordCount] = useState(0)
+
+  const [showName, setShowName] = useState('');
+  const [description, setDescription] = useState('');
+  const [tags, setTags] = useState([])
+
+  const [ticketPrice, setTicketPrice] = useState('');
+  const [ticketLink, setTicketLink] = useState('');
+  const [intermission, setIntermission] = useState(false);
+  const toggleIntermission = () => setIntermission(previousState => !previousState);
+  const [ticketRequired, setTicketRequired] = useState(false);
+  const toggleTicketRequired = () => setTicketRequired(previousState => !previousState);
+
+  const [venueName, setVenueName] = useState('');
+  const [venueAddress, setVenueAddress] = useState('');
+  const [wheelchair, setWheelchair] = useState(false)
+  const toggleWheelchair = () => setWheelchair(previousState => !previousState);
+  const [hearingAssistance, setHearingAssistance] = useState(false);
+  const toggleHearingAssistance = () => setHearingAssistance(previousState => !previousState);
+
+  const [cast, setCast] = useState([{name:'', role:''}]);
+  var castId = 0;
+
+  const [users, setUsers] = useState([]);
+  const currentUserInfo = useUser();
+  const updateUserInfo = useUserUpdate();
+  const currentUserFollowing = currentUserInfo.following;
+  const currentUserID = currentUserInfo.id;
+
+  const emptyState = () => {
+    setShowName('');
+    setDescription('');
+    setVenueName('');
+    setVenueAddress('');
+  };
+
+  const handlePress = () => {
+    if (!showName) {
+      Alert.alert('Show name is required!');
+    } else if (!venueName) {
+      Alert.alert('Venue name is required!');
+    } else if (!venueAddress) {
+      Alert.alert('Venue Address is required.');
+    } else if (wordCount > 250) {
+      Alert.alert('Description must be less than 250 words.')
+    } else {
+      handleAdd();
+      createEvent(
+        showName,
+        description,
+        venueName,
+        venueAddress,
+      );
+      navigation.navigate('Explore');
+      emptyState();
+      castId = 0;
+    }
+  };
+
+  const onChange = (t, e) => {
+    (e === 'role') ? (cast.length === castId ? setCast([...cast, {...(cast[castId]), role:t}]) : setCast([...(cast.slice(0,-1)), {...(cast[cast.length-1]), role:t}])) :
+    (cast.length === castId ? setCast([...cast, {...(cast[castId]), name:t}]) : setCast([...(cast.slice(0,-1)), {...(cast[cast.length-1]), name:t}]));
+  };
+  // FIX TO HANDLE EDITING PREVIOUS CAST STATES AFTER HANDLEADD() INCREMENTS CASTID PROB BY INCLUDING CASTID IN COMPONENT AND ACCESSING ON CLICK
+
+  // TODO:
+  //        1) ^ that by exporting cast as component
+  //        2) fix fetchUsers()
+  //        3) implement push to firebase
+
+  const handleAdd = () => {
```

Comment:
what if this added an empty { name, role } object into the end of the cast state rather than incrementing castId?